### ISSUE TITLE
Minishift brew cask out of date

### ISSUE THREAD

**Issue by tenfourty** [minishift/minishift#208]
Even though there has been a release of v1.0.0-beta.1 the latest version available via brew cask is 0.9.0.

Also, it's not possible to build from HEAD or devel using homebrew.

**Comment by hferentschik** [comment]
This is a duplicate of issue #161.

**Comment by hferentschik** [comment]
I guess what needs to happen here is to create a pull request for https://github.com/caskroom/homebrew-cask/blob/master/Casks/minishift.rb updating the cask definition. Most importantly the version needs updating, but also the packaging has changes. Instead of pure binaries, the release page now contains archive bundles (tgz, zip).

I am not a homebrew/cask user, so anyone up for trying to fix this cask definition and creating a pull request against caskroom/homebrew-cask. Most likely one need to create an issue in caskroom/homebrew-cask as well.

Some more info around creating and maintaining cask can be found here - https://github.com/caskroom/homebrew-cask/blob/master/doc/development/adding_a_cask.md

**Comment by hferentschik** [comment]
> Also, it's not possible to build from HEAD or devel using homebrew.

How come? What is the issue and how do you actually do this? As said, I am not a cask user.

**Comment by tenfourty** [comment]
Thanks for responding @hferentschik. I'm not sure this issue is a duplicate of #161, however I guess they both point to issues with the current brew install scripts.

> > Also, it's not possible to build from HEAD or devel using homebrew.
> 
> How come? What is the issue and how do you actually do this? As said, I am not a cask user.

Ideally the current brew scripts could be updated so that the default installs the current stable version. The brew devel option could install the latest unstable version (currently v1.0.0-beta.1) and HEAD could build from source from github. That's the usual pattern that folks follow.

In any case the release process for minishift should include updating homebrew install scripts so mac users can just do a brew upgrade to get the latest version when you do a release.

**Comment by jimmidyson** [comment]
@msavy Do you have any tips here?

**Comment by tenfourty** [comment]
@jimmidyson and @msavy another question while we are looking at the brew install process - why have you used `brew cask` instead of just a standard homebrew formula? I would have thought that `brew install minishift` would have been the most natural approach to this?
Depending on your answer I'd be happy to work on updating either the cask approach with a PR or creating a brew forumula.

**Comment by msavy** [comment]
Slow response as I'm on holiday :-).

One of the community members @lburgazzoli had submitted a formula before me, so I'm not actually involved, but I believe I can summarise as I tried to help out (please @lburgazzoli - jump in and clarify). 

- Because minishift is distributed as a binary it's not permitted in homebrew-core. Nowadays cask is shipped by default, so I don't see that it's a particularly severe impediment. 

- Furthermore, it would be difficult to submit as a non-binary as most if its constituent dependencies are not available in brew (including openshift itself). Naturally they also don't allow you to bypass the packaging system to pull down files, and they don't allow builds inside of a docker container [1].

- Minikube is distributed via Cask.

- It wasn't popular enough at the time to be allowed on homebrew-core anyway (some arbitrary metric involving stars/forks/watches which is now surpassed).

[1] If this analysis is incorrect then it may be possible to do a core formula and we can pull that thread further..

**Comment by msavy** [comment]
> Even though there has been a release of v1.0.0-beta.1 the latest version available via brew cask is 0.9.0.

Someone needs to make a PR to update it. 

- `brew cask edit minishift`
- Update and commit.
- Submit PR to cask (file should be at`/usr/local/Homebrew/Library/Taps/caskroom/homebrew-cask/Casks/minishift.rb`)

I would suggest just adding it to the release process. It's super simple by package manager standards. I can do this one manually if people want.

**Comment by msavy** [comment]
Done some digging. The standard cask tap does not accept non-final software.

You can have still a separate non-final formula (which will reside in caskroom/homebrew-versions).

E.g. minishift-beta or something. 

See: https://github.com/caskroom/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask

**Comment by tenfourty** [comment]
@msavy I can update this now... I think it's possible to include the -devel option for the unstable RCs etc.

**Comment by tenfourty** [comment]
Reading on... that is only with the main brew formulae and not with brew cask.

**Comment by lburgazzoli** [comment]
@msavy long story short, I had to jump between homebrew and homebrew-cask  a few time before being accepted in homebrew-cask, reason is as you said, binary formula are not accepted in homebrew and it is quite hard as today to build minishift from sources matching homebrew requirements.

So I guess until 1.0.0 get released, unstable/beta version should go to homebrew-versions

**Comment by tenfourty** [comment]
cool, I'm already working on a `minishift-unstable` version for homebrew-versions

**Comment by tenfourty** [comment]
Just FYI, I have a `minishift-unstable` PR ready to go for `homebrew-versions` once the above named issue is resolved. Source code is in my repo here https://github.com/tenfourty/homebrew-versions/blob/master/Casks/minishift-unstable.rb

**Comment by hferentschik** [comment]
@msavy, @tenfourty - awesome with this enthusiasm :-) 
Looking forward seeing this working again.

> In any case the release process for minishift should include updating homebrew install scripts so mac users can just do a brew upgrade to get the latest version when you do a release.

+1, maybe once the details on what has to, also in terms of stable vs beta releases, we can have a look at how this can be automated as part of the release process.

**Comment by tenfourty** [comment]
I've completed work on `minishift-unstable` and the PR is pending here https://github.com/caskroom/homebrew-versions/pull/3046 once that is accepted I'll create a PR for the minishift README.md to document how to get that version.

**Comment by msavy** [comment]
Just had a thought:

@hferentschik  @jimmidyson is it ever safe to have multiple versions of minishift coexisting in the same environment? I notice the metadata in .minishift doesn't seem to be versioned, so I'm not sure (NB: not asking about them being executed simultaneously).

If it's possible we could consider aliasing `minishift` to `minishift-beta` for the beta to avoid a clash with the standard installation.

It'd just be a case of changing: `binary 'minishift', target: 'minishift-beta'`

WDYT?

Ex:
```
[msavy@mmbp tmp]$ minishift version
minishift version: 0.9.0
[msavy@mmbp tmp]$ minishift-beta version
minishift version: 1.0.0-beta.1
```

**Comment by hferentschik** [comment]
>I've completed work on minishift-unstable and the PR is pending here caskroom/homebrew-versions#3046

awesome, thanks

> once that is accepted I'll create a PR for the minishift README.md to document how to get that version.

+1

**Comment by hferentschik** [comment]
> is it ever safe to have multiple versions of minishift coexisting in the same environment?

Sure, in most cases there should not really be an issue with that. It might be necessary to run _minishift delete_ between switching minishift versions, but that would depend on the differences. Generally the ~/.minishift directory contains cached data which should be save to keep and the actual VM files. 

> If it's possible we could consider aliasing minishift to minishift-beta for the beta to avoid a clash with the standard installation.

Not sure. I find this quite uncommon. Usually if you have multiple versions of a tool installed, it is up to you to configure the PATH variable according to your needs. Personally I would not like to think or use varying binary names.

**Comment by hferentschik** [comment]
Resolved by pull request #213
